kano_updater/status.py

```diff
@@ -125,12 +127,20 @@ def save(self):
         with open(self._status_file, 'w') as status_file:
             json.dump(data, status_file, indent=4)
 
-        # ensure splash is set whenever we are installing and not otherwise
-        if self._state in [self.INSTALLING_UPDATES,
-                           self.INSTALLING_INDEPENDENT]:
+        # When installing updates, configure the recovery stategy for the next
+        # boot in case of power failure. This sets a different splash screen
+        # during bootup and configures the system to autologin as the user in
+        # order to start the Updater immediately. See kano-ui-autostart.
+        if self._state in [self.INSTALLING_UPDATES, self.INSTALLING_INDEPENDENT]:
+            logger.debug('Configuring recovery stategy for next boot')
             set_splash_interrupted()
+            enable_console_autologin(get_user_unsudoed())
```

Comment:
So, the only concern I have here is whether any of these can fail. 
In particular `get_user_unsudoed()` might not always work.
The typical prescence of `/root/.kano-logs/kano-updater.log`  suggests that sometimes the updater is run in a context in which it can't figure out what the user is. Given that the state is modified in every context in which the updater runs, maybe this should be run in a try-except block.

Also - and this is my fault for originally putting the splash here - it' is a little obscure to have the state making these system changes. Perhaps it would be clearer if all of these calls were separated out into  functions like `updater_system_prepare` `updater_system_release` (or something like that) so it's more obvious.